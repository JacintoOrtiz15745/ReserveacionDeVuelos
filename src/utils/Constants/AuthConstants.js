const GoogleId = '555459120478-5dvb5sj1o0996s86gg8bk69ogm7vdiu4.apps.googleusercontent.com';  
export const AuthConstants = {
  GoogleId,
};
