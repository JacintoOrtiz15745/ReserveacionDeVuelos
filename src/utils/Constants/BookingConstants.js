const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next';
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';
const BookingToRoute = 'BookingTo';
const BookingNowRoute = 'BookingNow';
const MyFlightsRoute = 'MyFlights';
const BookingDateRoute = 'BookingDate';
const BookingDoneRoute = 'BookingDone';
const BookingPassengersRoute = 'BookingPassenger';
const doneMessage = 'Your request was received.';
const finish = 'Finish';
const PickerData = [
  {
    label: 'Amsterdam, Netherlands',
    value: 'AMS,Netherlands',
  },
  {
    label: 'Berlin, Germany',
    value: 'BER,Germany',
  },
  {
    label: 'Paris, France',
    value: 'PAR,France',
  },
  {
    label: 'London, UK',
    value: 'LON,UK',
  },
  {
    label: 'Madrid, Spain',
    value: 'MAD,Spain',
  },
];
const passengerNumberData = [
  {
    id: '1',
    title: '1',
  },
  {
    id: '2',
    title: '2',
  },
  {
    id: '3',
    title: '3',
  },
  {
    id: '4',
    title: '4',
  },
  {
    id: '5',
    title: '5',
  },
];
export const BookingConstants = {
  doneMessage,
  TitleOrigin,
  TitleDestination,
  TextInput,
  BtnText,
  SelectDate,
  Passengers,
  BookingToRoute,
  BookingNowRoute,
  MyFlightsRoute,
  passengerNumberData,
  PickerData,
  BookingDateRoute,
  BookingPassengersRoute,
  BookingDoneRoute,
  finish,
};
