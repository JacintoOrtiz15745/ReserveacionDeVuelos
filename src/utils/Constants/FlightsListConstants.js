const planeIconName = 'plane';
const passengers = 'passengers';
const myFlightsTitle = 'My flights';
const flightDataExperimental = [
  {
    id: '1',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'Serbia',
    destinyCity: 'Netherlands',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '2',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '3',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '4',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '5',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '6',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '7',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
];

export const FlightsListConstants = {
  planeIconName,
  passengers,
  myFlightsTitle,
  flightDataExperimental,
  
};
