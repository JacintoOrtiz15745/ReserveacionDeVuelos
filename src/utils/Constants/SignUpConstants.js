const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage =
  'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
const nameErrorMessage = ' must be at least 3 characters.';
const emailErrorMessage = 'invalid Email format';
const passwordErrorMessage = 'must be at least 8 characters';
const OR = 'or';
const SignIn = 'Sign In';
const SignUp = 'Sign Up';
const noAccountMsg = 'DonÂ´t have an account?';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
  nameErrorMessage,
  emailErrorMessage,
  passwordErrorMessage,
  OR,
  SignIn,
  noAccountMsg,
  SignUp,
};
