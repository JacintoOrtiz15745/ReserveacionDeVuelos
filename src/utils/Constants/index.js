
export * from './SignUpConstants';
export * from './FlightsListConstants'
export * from './AuthConstants'
export * from './BookingConstants'