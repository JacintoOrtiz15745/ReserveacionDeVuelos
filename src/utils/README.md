This folder will contain functions and utilities if needed.