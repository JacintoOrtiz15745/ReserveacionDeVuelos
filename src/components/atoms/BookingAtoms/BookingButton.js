import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {BookingConstants} from '../../../utils/Constants';
import {styles} from '../../../styles/BookingStyles';

const BookingButton = ({
  navigation,
  RouteToGo,
  selectedValue,
  values,
  date,
}) => {
  let paramsValue = {first: selectedValue};
  values &&
    values[1] &&
    (paramsValue = {
      first: values[0].first,
      second: values[1],
    });
  date && (paramsValue = [values[0], values[1], values[2]]);
  return (
    <TouchableOpacity
      testID="BookingBtn"
      style={styles.BookingButton}
      onPress={() => navigation.navigate(RouteToGo, paramsValue)}>
      <Text style={styles.BookingButtonText} testID="BookingBtnTitle">
        {BookingConstants.BtnText}
      </Text>
    </TouchableOpacity>
  );
};

export default BookingButton;
