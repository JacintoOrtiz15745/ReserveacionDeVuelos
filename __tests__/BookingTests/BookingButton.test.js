import React from 'react';
import {shallow} from 'enzyme';
import BookingButton from '../../src/components/atoms/BookingAtoms/BookingButton';
import {BookingConstants} from '../../src/utils/Constants';
describe('Booking Button render correctly', () => {
  const props = {
    navigation: {navigate: jest.fn()},
    RouteToGo: 'route',
    selectedValue: 'some value',
    values: {val1: '1', val2: '2'},
    date: false,
  };
  const wrapper = shallow(<BookingButton {...props} />);
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('component exists', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'BookingBtn',
    );
    expect(component.exists()).toBeTruthy();
  });
  test('title matches with our constant', () => {
    const title = wrapper.findWhere(
      (node) => node.prop('testID') === 'BookingBtnTitle',
    );
    expect(title.props().children).toEqual(BookingConstants.BtnText);
  });
  test('navigation function gets called as expected', () => {
    const btn = wrapper.findWhere(
      (node) => node.prop('testID') === 'BookingBtn',
    );
    btn.props().onPress();
    wrapper.update();
    expect(props.navigation.navigate).toHaveBeenCalled();
  });
});
