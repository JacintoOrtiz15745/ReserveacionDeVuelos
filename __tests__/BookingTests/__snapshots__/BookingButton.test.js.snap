// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Button render correctly snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#6472ee",
      "borderRadius": 10,
      "justifyContent": "center",
      "padding": 12,
    }
  }
  testID="BookingBtn"
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
    testID="BookingBtnTitle"
  >
    Next
  </Text>
</ForwardRef>
`;
